lib/scheduler.ml

```diff
@@ -428,23 +430,44 @@ module Make (Streamd : StreamDescriptor) = struct
           (* We check for activity again, because the stream may have gone
            * inactive after the call to `write` above. *)
           written, Streamd.requires_output descriptor
```

Comment:
I think we'll need to change this condition. A stream's subtree will be "active" if either: 
- its descriptor requires output
- the node itself has children

Reply:
Fixing a bad implementation around fixed responses revealed that this is the case in https://github.com/anmonteiro/ocaml-h2/pull/201